Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1985,12 +2040,78 @@ public void getShortUrl(int type, Collection<String> tags, String channel, Strin
      * @see BranchLinkData#putParams(String)
      * @see BranchLinkData#putDuration(int)
      * @see BranchLinkCreateListener
+     *
+     * @deprecated use {@link io.branch.referral.builders.BranchShortUrlBuilder} instead.
      */
+    @Deprecated
     public void getShortUrl(Collection<String> tags, String channel, String feature, String stage, JSONObject params, int duration, BranchLinkCreateListener callback) {
         generateShortLink(null, LINK_TYPE_UNLIMITED_USE, duration, tags, channel, feature, stage, stringifyParams(filterOutBadCharacters(params)), callback, true);
     }
 
     /**
+     * <p>Configures and requests a short URL to be generated by the Branch servers.</p>
+     *
+     * @param alias    Link 'alias' can be used to label the endpoint on the link.
+     *                 <p/>
+     *                 <p/>
+     *                 For example:
+     *                 http://bnc.lt/AUSTIN28.
+     *                 Should not exceed 128 characters
+     * @param type     An {@link int} that can be used for scenarios where you want the link to
+     *                 only deep link the first time.
+     * @param duration An {@link int} the time that Branch allows a click to remain outstanding
+     *                 and be eligible to be matched with a new app session.
+     * @param tags     An iterable {@link Collection} of {@link String} tags associated with a deep
+     *                 link.
+     * @param channel  A {@link String} denoting the channel that the link belongs to. Should not
+     *                 exceed 128 characters.
+     * @param feature  A {@link String} value identifying the feature that the link makes use of.
+     *                 Should not exceed 128 characters.
+     * @param stage    A {@link String} value identifying the stage in an application or user flow
+     *                 process. Should not exceed 128 characters.
+     * @param params   A {@link JSONObject} value containing the deep linked params associated with
+     *                 the link that will be passed into a new app session when clicked
+     * @param callback A {@link BranchLinkCreateListener} callback instance that will trigger
+     *                 actions defined therein upon receipt of a response to a create link request.
+     * @param  async   A {@link Boolean} whose value is true if the link should be created asynchronously.
```

Comment:
I'm not a fan of this method. It doesn't have to be removed now, but the fact that it has different return types *and* different control flows depending on a flag is offputting to say the least. Deprecated, so that's good at least..